### DISCUSSION TITLE
Issue with laser driver PWM button

### DISCUSSION THREAD

**Discussion by 4noxx** [arkypita/LaserGRBL#1291] [Q&A] [answered]
hello

i ve a problem to start my laser with lasergrbl. 

my connection:
![2021-01-19 18_46_05-Fotos](https://user-images.githubusercontent.com/12627059/105073004-be508e80-5a86-11eb-86bc-229b106ce584.jpg)

![2021-01-19 18_51_15-Foto - Google Fotos](https://user-images.githubusercontent.com/12627059/105073434-5d758600-5a87-11eb-9b0d-e29ddfe153ad.jpg)

1) is this right?

i started laserGRBL, install GRBL 1.1f with Tools / Install GRBL Firmware
connected to the Arduino / press focus button..... but nothing happens

over the push button on the laser, i can activated the laser. but not with the software.....

my laser:
https://www.amazon.de/gp/product/B08ML7G1RT

**Comment by arkypita** [comment]
This is not a LaserGRBL issue, and should be posted under [discussion](https://github.com/arkypita/LaserGRBL/discussions) session.
(moved)

**Comment by arkypita** [comment]
Wiring is correct. Maybe your laser driver need an higher S value to turn on. Try editing (right click, edit) the focus button.

**Reply by 4noxx** [reply]
hmm, thx. 

i will try it

here are the laserGRBL settings

![2021-01-19 19_12_46-Window](https://user-images.githubusercontent.com/12627059/105076213-0f628180-5a8b-11eb-8ff9-594b46166437.jpg)

![2021-01-19 19_13_10-Window](https://user-images.githubusercontent.com/12627059/105076211-0ec9eb00-5a8b-11eb-8afd-b5a638b538a0.jpg)

**Comment by arkypita** [comment]
Settings are ok.
I don't know where your custom button come from. Can you take a screenshot of the laser focus button gcode? (right click, edit)

**Comment by 4noxx** [comment]
i had also tried the original buttons. same result
![2021-01-19 19_53_41-LaserGRBL v4 0 5](https://user-images.githubusercontent.com/12627059/105079732-1dff6780-5a90-11eb-95cb-26ee122ffc52.jpg)

**Comment by arkypita** [comment]
$30 * 3 / 100 mean 3% of power (S7 if $30 = 255)
Try set $30 * 10 / 100

See if you find something usefull here:

https://github.com/arkypita/LaserGRBL/issues/1165
https://github.com/arkypita/LaserGRBL/issues/566
https://github.com/arkypita/LaserGRBL/issues/368

**Comment by 4noxx** [comment]
thx, i read this, but the laser dont turn on

i test the cnc shield with a led, if i push the focus button, the led goes on....

![led](https://user-images.githubusercontent.com/12627059/105152148-1a5af780-5b07-11eb-85d4-3a8d5aa150fd.jpg)

**Comment by arkypita** [comment]
Your test is a good test.

So the issue is on the laser driver side.
If you connect driver PWM wire to 5V pin into CNC v3 shield (yellow header), does the laser turn ON or not? (be care: wear goggles, point out from you!)

Can you put a detailed picture of the driver board?
Is there any logo/text in particular that can make it clear which model the driver is?

**Comment by 4noxx** [comment]
_**If you connect driver PWM wire to 5V pin into CNC v3 shield (yellow header), does the laser turn ON or not? (be care: wear goggles, point out from you!)**_

No, only the LED burns.


_**Can you put a detailed picture of the driver board?**_

here 2 pics, one with a burned test led (focus on)

![board](https://user-images.githubusercontent.com/12627059/105168821-f4d7e900-5b1a-11eb-883e-45b8d79358dc.jpg)
![board_led](https://user-images.githubusercontent.com/12627059/105168825-f5707f80-5b1a-11eb-9c65-86093f46bbc2.jpg)

looks like this here:
https://de.aliexpress.com/item/4000753151351.html
or
https://de.aliexpress.com/item/4000753344114.html

**Comment by arkypita** [comment]
>> If you connect driver PWM wire to 5V pin into CNC v3 shield (yellow header), does the laser turn ON or not? (be care: wear goggles, point out from you!)
> 
> No, only the LED burns.

This suggests your driver's PWM control is gone.

**Comment by arkypita** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/8782035/105175426-2ef9b880-5b24-11eb-8095-2f91a79970d5.png)


This instead makes me think that to have PWM control the button must be pressed.
How is your button?

**Comment by arkypita** [comment]
> > If you connect driver PWM wire to 5V pin into CNC v3 shield (yellow header), does the laser turn ON or not? (be care: wear goggles, point out from you!)
> 
> No, only the LED burns.

Please, repeat the test BUT with ON/PWM switch pressed.
Connect to +5, and then connect to GND. Let me know what happens

**Comment by 4noxx** [comment]
sorry, you are right. problem was the "push button". this is a switch for PWM on/off.

works now...

am ashamed
